### PR TITLE
Git module validation

### PR DESCRIPTION
**Pull request by logicminds**
This adds the capability to verify git based modules as I have previously done with the ra10ke validation task.  I figured it would be wiked cool to have vscode validate a puppetfile git urls and refs too.

This doesn't add any actual validation but just the ability to add later since I don't really know the codebase and how it all comes together.

### CODE REVIEW

File: lib/puppetfile-resolver/puppetfile/git_module.rb [hunk 1]
```diff
@@ -10,10 +10,89 @@ class GitModule < BaseModule
       attr_accessor :commit
       attr_accessor :tag
 
+      alias :url :remote
+
+      REMOTE_REFS_CMD = 'git ls-remote --symref'
```
Comment (glennsarti) [inline lib/puppetfile-resolver/puppetfile/git_module.rb:16]: I wonder, is it worth checking if `git` is even available?  Not everyone may have it on the PATH.

File: Rakefile [hunk 2]
```diff
@@ -1,4 +1,25 @@
 # frozen_string_literal: true
+require 'rubygems'
+require 'bundler'
+require 'bundler/gem_tasks'
+require 'rake/testtask'
```
Comment (glennsarti) [inline Rakefile:5]: Thankyou!!

**Comment by glennsarti** [comment]
Overall this looks really good!  Only question remains to figure out what to do if git isn't available.

File: lib/puppetfile-resolver/puppetfile/git_module.rb (hunk 1 above)
Comment (logicminds) [reply lib/puppetfile-resolver/puppetfile/git_module.rb:16]: If this is the case I would argue for some util libraries for running commands. Then those could search for git and possibly raise an error if not found.

**Comment by glennsarti** [comment]
Closing in favour of #33

**Pull request closed without merging**